File: Frontend/src/api/types/Movie.ts
import { z } from 'zod';

export const MovieSchema = z.object({
    id: z.string(),
    authorName: z.string(),
    title: z.string(),
    description: z.string(),
    rating: z.number(),
    releaseDate: z.string(),
    director: z.string(),
    duration: z.string(),
    cast: z.string(),
    background: z.string(),
    events: z.object({
        id: z.string(),
        title: z.string(),
        start: z.string(),
        end: z.string(),
        availableSeats: z.number(),
        price: z.number(),
        description: z.string(),
    }).array()
});

export const MoviesSchema = z.object({
    id: z.string(),
    authorName: z.string(),
    title: z.string(),
    description: z.string(),
    rating: z.number(),
    background: z.string(),
}).array();

export type Movie = z.infer<typeof MovieSchema>;
export type Movies = z.infer<typeof MovieSchema>[]
export type Event = z.infer<typeof MovieSchema>['events'][0]

File: Backend/Backend-Cinema/WebApi/Models/Movie.cs
namespace WebApi.Models;

public class Movie
{
    public Guid Id { get; set; }
    public string AuthorName { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public double Rating { get; set; }
    public string Background { get; set; }
    public string ReleaseDate { get; set; }
    public string Director { get; set; }
    public string Duration { get; set; }
    public string Cast { get; set; }
    
    public List<Event> Events { get; set; }
}